fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/NumericAggregationValue.java

```diff
@@ -591,8 +672,37 @@ public enum PhysicalOperator {
         MAX_I(LogicalOperator.MAX, TypeCode.INT, TypeCode.INT, Objects::requireNonNull, (s, v) -> Math.max((int)s, (int)v), identity()),
         MAX_L(LogicalOperator.MAX, TypeCode.LONG, TypeCode.LONG, Objects::requireNonNull, (s, v) -> Math.max((long)s, (long)v), identity()),
         MAX_F(LogicalOperator.MAX, TypeCode.FLOAT, TypeCode.FLOAT, Objects::requireNonNull, (s, v) -> Math.max((float)s, (float)v), identity()),
-        MAX_D(LogicalOperator.MAX, TypeCode.DOUBLE, TypeCode.DOUBLE, Objects::requireNonNull, (s, v) -> Math.max((double)s, (double)v), identity());
-
+        MAX_D(LogicalOperator.MAX, TypeCode.DOUBLE, TypeCode.DOUBLE, Objects::requireNonNull, (s, v) -> Math.max((double)s, (double)v), identity()),
+        /*
+        pair.key -> last aggregation result, initial = 0
+        pair.value -> input value to be aggregated
+        so final result = pair.key | (1 << pair.value)
+        */
+        BITMAP_LL(LogicalOperator.BITMAP, TypeCode.LONG, TypeCode.BYTES,
+                s -> {
+                    BitSet bs = new BitSet();
+                    bs.set(((Long)s).intValue());
+                    return bs.toLongArray()[0];
+                },
+                (s, v) -> {
+                    // convert s (long to bitset), set v, then convert it back to long
+                    BitSet bs = BitSet.valueOf(new long[] {(long)s});
+                    bs.set(((Long)v).intValue());
+                    return bs.toLongArray()[0];
```

Comment:
I am not sure how this is supposed to work, it seems we're limiting the bit set's size to 64 bits only, am I correct?

Can you please illustrate via an example how the aggregation should work?

I imagine if this is used in the context of a streaming aggregate, we would have to group the rows according to some bucket size, for example:

```sql
SELECT BITMAP(C)
FROM T
GROUP BY BIT_BUCKET(C, 1000)
```

Let's say we have the following rows:
```sql
C
==
0
1
4
64
65
1001
1002
---
```

I expect to get two rows back representing two groups:
```sql
RESULT
======
[1 1 0 0 1 ...... 0] [1 1 0 .... 0] <-- each representing a single byte, first row contains two longs, sort of VLE.
[0 1 1 0 .......  0]                <-- second group, mapping bits starting at 1000-inclusive.
```

If that's the case, and unless I am missing something, these aggregate definitions (initial, partial, and final) are incorrect because they do not take into consideration the actual value ranges and only assume 64-bit wide range

Reply:
The bits in the bitmap are little-endian. But the byte order isn't reversed. Which means 1,2,3 is [0x0E, ...(124 zeros)].